modules/test/playwright/tests/object-web/objectRelationships.spec.ts

```diff
@@ -258,4 +259,119 @@ test.describe('Manage object relationships through Model Builder', () => {
+				objectFolderExternalReferenceCode:
+					objectFolder.externalReferenceCode,
+				status: {code: 0},
+			});
+
+		const objectDefinition2 =
+			await apiHelpers.objectAdmin.postRandomObjectDefinition({
+				objectFields: [],
+				objectFolderExternalReferenceCode:
+					objectFolder.externalReferenceCode,
+				status: {code: 1},
+			});
+
+		createdEntities.objectDefinitionIds.push(
+			objectDefinition1.id,
+			objectDefinition2.id
+		);
+
+		const objectRelationshipLabel =
+			'objectRelationshipLabel' + getRandomInt();
+		const objectRelationshipName =
+			'objectRelationshipName' + Math.floor(Math.random() * 99);
+
+		const objectRelationshipData: Partial<ObjectRelationship> = {
+			label: {
+				en_US: objectRelationshipLabel,
+			},
+			name: objectRelationshipName,
+			objectDefinitionExternalReferenceCode1:
+				objectDefinition1.externalReferenceCode,
+			objectDefinitionExternalReferenceCode2:
+				objectDefinition2.externalReferenceCode,
+			objectDefinitionId1: objectDefinition1.id,
+			objectDefinitionId2: objectDefinition2.id,
+			objectDefinitionName2: objectDefinition2.name,
+			type: 'oneToMany' as ObjectRelationshipType,
+		};
+
+		await apiHelpers.objectAdmin.postObjectRelationship(
+			objectRelationshipData
+		);
+
+		const publishedObjectDefinition2 =
+			await apiHelpers.objectAdmin.postObjectDefinitionPublish(
+				objectDefinition2.id
+			);
+
+		await viewObjectDefinitionsPage.goto();
+
+		await viewObjectDefinitionsPage.openObjectFolder(
+			objectFolder.label['en_US']
+		);
+
+		await viewObjectDefinitionsPage.viewInModelBuilderButton.click();
+
+		await modelBuilderPage.fitViewButton.click();
+
+		await modelBuilderPage.clickObjectRelationshipEdge(
+			objectRelationshipLabel
+		);
+
+		await modelBuilderPage.deleteObjectRelationship(
+			objectRelationshipData.name
+		);
+
+		await expect(modelBuilderPage.deletionNotAllowed).toBeVisible();
+
+		const objectFieldObjectRelationship =
+			publishedObjectDefinition2.objectFields.find(
+				(objectField: ObjectField) =>
+					objectField.businessType === 'Relationship'
+			);
+
+		await expect(
+			page.getByText(
+				`The object field "${objectFieldObjectRelationship.name}" cannot be deleted because it is the only custom object field of the published object definition "${objectDefinition2.name}". Add at least one object field to the object definition.`
+			)
+		).toBeVisible();
+
+		await page.getByRole('button', {name: 'Done'}).click();
+
+		await modelBuilderPage.createObjectField({
+			mandatory: false,
+			objectDefinitionName: objectDefinition2.name,
+			objectFieldBusinessType: 'Text',
+			objectFieldLabel: 'objectField' + getRandomInt(),
+		});
+
+		await modelBuilderPage.clickObjectRelationshipEdge(
+			objectRelationshipLabel
+		);
+
+		await modelBuilderPage.deleteObjectRelationship(
+			objectRelationshipData.name
+		);
+
+		await waitForSuccessAlert(
+			page,
+			'Success:Relationship was deleted successfully.'
+		);
+	});
```

Comment:
hey Igor, here we could just add a field to objectDefinition2 via API, and then the clean up of the relationship, definitions and folder will work as expected!